docs/mapalgebra/colors.md

```diff
@@ -0,0 +1,182 @@
+default rendering options due to the fact that certain indices (like
+NDVI) have scales which are meaningful regardless of a raster layer's
+actual values (in other words, we don't need to assume that the
+distribution calculated by way of a generated histogram accurately
+conveys the significance of a layer's values).
+
+
+Minimal `NodeMetadata` specifying color breaks:
+```json
+{
+  "label": null,
+  "description": null,
+  "histogram": null,
+  "colorRamp": null,
+  "classMap": null,
+  "breaks": [-1.0, -0.8, -0.6000000000000001, -0.4000000000000001,
+-0.20000000000000007, -5.551115123125783E-17, 0.19999999999999996,
+0.39999999999999997, 0.6, 0.8, 1.0]
+}
+```
+
+
+
+### Default and override metadata
+Each node of an AST can specify default `NodeMetadata` (or not).
+`ToolRun`s have the ability to override these defaults by providing
+a mapping from the node ID to the preferred `NodeMetadata` values.
+
+Here's an NDVI `Tool` definition with a sensible (if imprecise) set of
+color breaks:
+```json
+{
+  "apply": "/",
+  "args": [{
+    "apply": "-",
+    "args": [{
+      "id": "c583c503-336b-40ac-9405-e37988fbfaa0",
+      "metadata": null
+    }, {
+      "id": "2ca64844-6783-42ae-b953-ebde1d2d0f1b",
+      "metadata": null
+    }],
+    "id": "968ba2d0-4740-4179-83f9-d229c609a62d",
+    "metadata": {
+      "label": "c583c503-336b-40ac-9405-e37988fbfaa0",
+      "description": null,
+      "histogram": null,
+      "colorRamp": null,
+      "classMap": null,
+      "breaks": null
+    }
+  }, {
+    "apply": "+",
+    "args": [{
+      "id": "c583c503-336b-40ac-9405-e37988fbfaa0",
+      "metadata": null
+    }, {
+      "id": "2ca64844-6783-42ae-b953-ebde1d2d0f1b",
+      "metadata": null
+    }],
+    "id": "1ff78cb5-3aa0-4c39-b85d-f7139a138a1c",
+    "metadata": {
+      "label": "c583c503-336b-40ac-9405-e37988fbfaa0",
+      "description": null,
+      "histogram": null,
+      "colorRamp": null,
+      "classMap": null,
+      "breaks": null
+    }
+  }],
+  "id": "5817ce54-d701-493f-9420-426ec512f1a0",
+  "metadata": {
+    "label": "c583c503-336b-40ac-9405-e37988fbfaa0",
+    "description": null,
+    "histogram": null,
+    "colorRamp": null,
+    "classMap": null,
+    "breaks": [-1.0, -0.8, -0.6000000000000001, -0.4000000000000001, -0.20000000000000007, -5.551115123125783E-17, 0.19999999999999996, 0.39999999999999997, 0.6, 0.8, 1.0]
+  }
+}
+```
+
+If we wanted to override it with a more precise set of breaks, this would work:
+```json
+{
+  "sources": {
+    "c583c503-336b-40ac-9405-e37988fbfaa0": {
+      "id": "b501307c-8cae-45ec-8b49-9229481f0457",
+      "band": 4,
+      "type": "project"
+    },
+    "2ca64844-6783-42ae-b953-ebde1d2d0f1b": {
+      "id": "c6dbcd52-852f-4fa0-9dc4-e4385b653680",
+      "band": 5,
+      "type": "project"
+    }
+  },
+  "metadata": {
+    "5817ce54-d701-493f-9420-426ec512f1a0": {
+      "label": "c583c503-336b-40ac-9405-e37988fbfaa0",
```

Comment:
The circe `Decoder` for `Option` assumes missing fields are `None` rather than failure. You should definitely be able to omit those extra `null` fields